src/main/java/com/telenordigital/sms/smpp/SubAddress.java

```diff
@@ -20,10 +20,22 @@
  * #L%
  */
 
+import java.util.Optional;
+
 /**
  * For source_subaddress or dest_subaddress in deliver_sm
  *
  * @param mcc Mobile country code
  * @param mnc Mobile network code
  */
-public record SubAddress(int mcc, int mnc) {}
+public record SubAddress(int mcc, int mnc) {
+  public static Optional<SubAddress> create(final String input) {
```

Comment:
may be change the method name to parse?

Reply:
Sure, I'll change the name. We do get SubAddresses with no information, just a single byte. It is normal, but it doesn't give us anything, so I think we should not log, and return empty.